### DISCUSSION TITLE
The appearance of TextBox

### DISCUSSION THREAD

**Discussion by joyhooian** [kikipoulet/SukiUI#231] [Q&A] [answered]
Hi, I use the TextBox in my demo code and it was found that not same with the TextBox in the Document and the Gallery App.
Is there any breaking changes on the default appearance/style of TextBox? Or I need to use some "style/classes" attr. to change the appearance?
#### Appearance in Gallery
![image](https://github.com/kikipoulet/SukiUI/assets/27796862/7b69b58a-da39-4904-a4d8-088bc73f218c)
#### Appearance in my demo
![image](https://github.com/kikipoulet/SukiUI/assets/27796862/f5b935f5-3767-4598-9c2e-5df871f0051f)
#### TextBox demo code
![image](https://github.com/kikipoulet/SukiUI/assets/27796862/e0035d3e-3f5e-4104-b29f-cc2e656fb4dc)

**Comment by AuroraZiling** [comment]
Hi,
Currently, the current default TextBox design is what you see in your demo.
There are some classes:
![image](https://github.com/user-attachments/assets/990ea05d-696c-4e25-8c8b-39a17c3756cc)

**Comment by kikipoulet** [comment, marked as answer]
Please note that the current design is temporary. Textbox and variants are the last major design change that I want to make in the library.

I'm quite unsatisfied with the past and current form design and I wish I had a more impactful idea. Something that would be minimalist, light and incorporated perfectly in the almost-glass aspect of the library. With more interactivity, like an optional title animation to make form design easier. But I'm still waiting for that inspiration to come.

So expect it to change one day 💡

**Comment by kikipoulet** [comment]
@AuroraZiling 

Small update, thinking about using the glasscard for textbox, look better than I expected actually

![image](https://github.com/user-attachments/assets/2b3fe356-140b-4d5b-a714-47b4fd96a3e9)

As a side note, I'm waiting to find a way to have a "custom" cornerradius implementation like I mentionned in this comment : https://github.com/kikipoulet/SukiUI/pull/240#issuecomment-2203984127

I really need to create this for button, textbox, glasscard, ..  It seems like a detail but it would make a nice final difference.

Edit : Well, tell me what you think about it but replacing the plain background of components by a glasscard (progressbar, switch and slider on this screenshot) look nice, isn't it ? 

![image](https://github.com/user-attachments/assets/23b03fd1-4ee0-4554-9704-8bbfc5bae27a)

**Comment by AuroraZiling** [comment]
looks great, but I had a small concern about performance (due to the more widespread use of GlassCard)

**Reply by kikipoulet** [reply]
> looks great, but I had a small concern about performance (due to the more widespread use of GlassCard)

Actually I tried to modifiy GlassCard control to just a controlpresenter, so removing all the glasscard for testing purpose and :

SukiDemo with GlassCard : 117 Mo RAM
SukiDemo without GlassCard : 110 Mo RAM

So even if doing it add 2-3 mo RAM it's fine okay right ? 😵

**Reply by AuroraZiling** [reply]
ok, fine😀
